### PR TITLE
Make it easier to install extension on Alpine Linux

### LINKED ISSUE #19: Building on Alpine Linux is overcomplicated
Building for Alpine Linux requires that we [set the `LDFLAGS` environment variable](https://github.com/RubixML/Tensor/blob/2.2.0/.github/workflows/ci-ext.sh#L13).

What about improving the `config.m4` file to simplify users' life, so that they can simply call `pecl install tensor` without any hassle?

### PR DESCRIPTION
**Pull request by mlocati**
Fix #19

### CODE REVIEW

**Review by andrewdalpino — COMMENTED** [review]
Nice @mlocati ! How is the patch supposed to be run now that composer won't be able to find the patch script when we call `$ composer compile`?

https://github.com/mlocati/Tensor/blob/build-ext-fixes/composer.json#L70

Should we run the `build-ext` script from this command instead?

File: build-ext [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+chdir(__DIR__);
+
+passthru('zephir generate', $rc);
+if ($rc !== 0) {
+    exit(1);
+}
+
+$configM4Contents = file_get_contents(CONFIG_M4_PATH);
+if (!$configM4Contents) {
```
Comment (andrewdalpino) [inline build-ext:27]: Loving these added checks

File: build-ext [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	if test $have_backtrace_symbols = no; then
+		LDFLAGS="${LDFLAGS:-} -lexecinfo"
+	fi
+
+EOT
+;
+
+chdir(__DIR__);
+
+passthru('zephir generate', $rc);
```
Comment (andrewdalpino) [inline build-ext:21]: Will this only generate the C source? What happens if we want to compile the whole extension from Zephir and C source like we currently do with `$ composer compile`?

**Comment by mlocati** [comment]
> How is the patch supposed to be run now that composer won't be able to find the patch script when we call `$ composer compile`?

Whoops, I missed that composer script!

BTW, I think that https://github.com/RubixML/Tensor/blob/2.2.0/composer.json#L68-L69 is logically wrong.
1. `zephir build --no-dev` is the same as calling these commands in sequence:
    1. `zephir generate`
    2. `zephir compile --no-dev`
    3. `zephir install --no-dev`
2. `extension-config-patch.php` is meant to patch the result of `zephir generate` before the extension gets compiled with `zephir compile`

So, at the moment (version 2.2.0), we patch `config.m4` *after* it's being used.

I've updated the `composer compile` script so that we patch `config.m4` *before* it's being used.

File: build-ext (hunk 2 above)
Comment (mlocati) [reply build-ext:21]: `build-ext` is meant to be a wrapper of `'zephir generate` only.

**Review by andrewdalpino — APPROVED** [review]
There we go! Thank you so much @mlocati

**Pull request merged**